crates/fee/src/lib.rs

```diff
@@ -480,13 +483,17 @@ impl<T: Config> Pallet<T> {
         nominator_id: &T::AccountId,
         currency_id: CurrencyId<T>,
     ) -> Result<BalanceOf<T>, DispatchError> {
-        frame_support::storage::with_transaction::<_, DispatchError, _>(|| {
-            sp_runtime::TransactionOutcome::Rollback(Self::distribute_and_compute_vault_rewards(
-                vault_id,
-                nominator_id,
-                currency_id,
-            ))
-        })
+        // use a closure so we can use the `?` operator
+        let get_rewards = || -> Result<BalanceOf<T>, DispatchError> {
+            let balance_before = currency::get_free_balance::<T>(currency_id, nominator_id);
+            Self::withdraw_vault_rewards(vault_id, nominator_id, None, currency_id)?;
+            let balance_after = currency::get_free_balance::<T>(currency_id, nominator_id);
+            let reward = balance_after.saturating_sub(&balance_before)?;
+            Ok(reward.amount())
+        };
+
+        // don't commit storage changes
+        storage::with_transaction(|| TransactionOutcome::Rollback(get_rewards()))
```

Comment:
Fwiw I don't think we need to do this, at least if it is only consumed within the RPC. You can see that I didn't do that here: https://github.com/interlay/interbtc/pull/842

Reply:
noted, we can refactor later, but merging for now